Include backRoute & context in content link on BrowseResourceMetadata

nucleogenesis opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

## Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->

Two bugs fixed.

1. When accessing Bookmarked content from Bookmark's page's *side panel* the route of the content's back arrow would be the Learn Home page.
2. Also if a user would have search context and access the content via the side panel, then they'd have to click the back arrow twice and get sent to Learn Home

This change ensures correct context in both cases.

## References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
--> 

Fixes #8830 

## Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

First - confirm the two bugs above on release-0.15

This  is only when clicking view resource in the side panel when browsing content.

Try it in Bookmarks

Try it in the Library Page - with and without search criteria in place. It should always go back to the state you were in when you left.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

nucleogenesis committed:
Include backRoute & context in content link on BrowseResourceMetadata

rtibbles reviewed (approved):
Code makes sense - bookmark backlinks work!

nucleogenesis committed:
mount a router in browse-resource-metadata spec

with the change to ensure context is kept across navigation, BrowseResourceMetadata calls upon $route - so this makes sure it's present properly
